bmap-writer.cpp

```diff
@@ -26,29 +26,69 @@
 #include <string>
 #include <chrono>
 #include <cstring>
+#include <cerrno>
+
 #include <fcntl.h>
+#include <getopt.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <errno.h>
+
 #include <libxml/parser.h>
 #include <libxml/tree.h>
 #include <openssl/evp.h>
 #include <archive.h>
 
-#define CHECKSUM_LENGTH 64
-#define RANGE_LENGTH    19
-
 struct range_t {
     std::string checksum;
-    std::string range;
+    size_t startBlock;
+    size_t endBlock;
 };
 
 struct bmap_t {
     std::vector<range_t> ranges;
     size_t blockSize;
 };
 
+enum ChecksumMode {
+    NO_CHECKSUM,
+    CHECKSUM_READ,
+    CHECKSUM_WRITTEN,
+};
+
+class Checksummer {
```

Comment:
I would like to avoid using class object oriented code here.

Reply:
Ok, I'll try to rewrite it using a `struct` + init/deinit functions.

Reply:
Done in updated MR.